### DISCUSSION TITLE
Implementing RedPajama models

### DISCUSSION THREAD

**Discussion by georgepullen** [abetlen/llama-cpp-python#413] [Q&A]
Hello, Is there a way I can replace the main.exe file (that is used for inference with LLaMa when you run ./main) with the redpajama.exe file?

**Comment by gjmulder** [comment]
That's not how it works. `main` is provided by `llama.cpp`. `llama-cpp-python` provides a Python interface to `llama.cpp`.

[HuggingFace](https://huggingface.co/models?search=ggml) provides access to a wide range of [GGML](https://github.com/ggerganov/ggml) models that are mostly compatible with `llama.cpp` and therefore `llama-cpp-python`.

[Some of these models](https://huggingface.co/openlm-research) were trained using open source datasets such as [Red Pajama](https://www.together.xyz/blog/redpajama). Most are based on [Facebook's Llama models](https://ai.facebook.com/blog/large-language-model-llama-meta-ai/) and are therefore [distributed under a restricted license](https://docs.google.com/forms/d/e/1FAIpQLSfqNECQnMkycAp2jP4Z9TFX0cGR4uf7b_fBxjY_OjhJILlKGA/viewform?pli=1).

**Reply by georgepullen** [reply]
Sorry, I should clarify I'm referring to Together Computer's INCITE 3b and 7b models. 

In regards to the ```redpajama.exe``` file, i'm referring to the one in the ```redpajama.cpp``` fork of ```llama.cpp```. Since the ```redpajama.exe``` file works as a drop in replacement for the ```main.exe``` file, I was wondering if I could somehow replace it in ```llama-cpp-python``` in order to get python bindings working for ```redpajama.cpp```.
